src/nextjs/server/routeMatcher.ts

```diff
@@ -52,7 +47,10 @@ export type RouteMatcherParam =
  * predefined routes that can be passed in as the first argument.
  *
  * You can use glob patterns to match multiple routes or a function to match against the request object.
- * Path patterns and regular expressions are supported, for example: `['/foo', '/bar(.*)'] or `[/^\/foo\/.*$/]`
+ * Path patterns are supported, for example: `['/foo', '/bar{/*extra}']`.
+ *
+ * Note: regular expressions are not supported in path patterns.
+ *
  * For more information, see: https://github.com/pillarjs/path-to-regexp
  */
 export const createRouteMatcher = (routes: RouteMatcherParam) => {
```

Comment:
You can write type level tests for stuff like this.https://github.com/get-convex/convex-js/blob/main/src/server/schema.test.ts is a fancier example, but seems like it would be nice to test
```
createRouteMatcher("/foo.*")
```
doesn't compile if that's the intent.

One way of doing this is adding a `// @ts-expect-error` comment above the line that's meant to fail typechecking (compilation will succeed if the next line has a type error, and fail otherwise).